### DISCUSSION TITLE
Is v4 strictly not recommended for production use?

### DISCUSSION THREAD

**Discussion by NicoEssi** [isar/isar#1456] [Help]
Truly amazing work with the project, really impressed with it and I'm excited to work with it.

I'd like to understand what the precautions are regarding using latest version of dev for production. I ask because I need the string IDs, encrypted database, and support for List<dynamic> for my use-case.

My understanding is that v4 is not ready for production because there is no migration capability from v3. Is this all?

Alternatively, can I safely start working with v4 assuming its stable release is due sometime this month or November?

**Comment by Flajt** [comment]
Have a look at [this](https://github.com/isar/isar/discussions/1412). You can use it for developement apperantly.

**Reply by NicoEssi** [reply]
Unfortunately there's no information as to why it is ready for development but not production.

**Comment by tschiekdev** [comment]
Is the project somehow progressing? Or is that dead? Is there a roadmap when v4 will be ready?

**Reply by Flajt** [reply]
I don't think it's dead from my observations. It's just one busy person managing the project so I assume everything is ok. 

If you have a look at the release cycle it sees be matching a pattern so it's fine I think.

**Comment by tschiekdev** [comment]
"One" person??? :-(

It looks like I have to look for something else. How could I use something like this productively if you can't expect reliable support for a long time.

Really sad that Flutter/Dart doesn't bring anything of its own on the part of Google.

**Reply by Flajt** [reply]
I mean it's opensouce, it's always depending on the people, for this reason feel free to contribute. 

You could also opt for firebase, realmsdb or objectbox. And if you want you can replace it later on. Or have a look at couchdb. 

There is a great [website](https://fluttergems.dev/) which keeps track of different tools in the flutter system. Maybe you find something that helps you. (Check the db section)

**Comment by jason-downs** [comment]
v4 preview release has been sitting pretty for several months now(4.0.0-dev.14) without any updates....can we expect an official Isar v4 release soon? I'm starting to get worried.

**Reply by Bohne13** [reply]
Simon did merge some test yesterday, but beside that, it's pretty quiet.

We have to have in mind, it's an open source project with mostly Simon as single developers in his free time.

He will let us know, when it's ready.

**Reply by Flajt** [reply]
There seems also to be a telegram channel for the project. Maybe you can gather some insides there.
